Packs/MISP/Integrations/MISPV3/MISPV3.py

```diff
@@ -0,0 +1,1411 @@
+    limit = demisto_args.get('limit')  # type: str
+    limit_int = int(limit) if limit.isdigit() else 0
+    osint_url = f'{url}/manifest.json'
+    not_added_counter = 0
+    try:
+        uri_list = requests.get(osint_url, verify=use_ssl, headers=headers, proxies=proxies).json()
+        events_numbers = list()  # type: List[Dict[str, int]]
+        for num, uri in enumerate(uri_list, 1):
+            req = requests.get(f'{url}/{uri}.json', verify=use_ssl, headers=headers, proxies=proxies).json()
+            e = MISPEvent()
+            e.load(req)
+            event = PYMISP.add_event(e)
+            event_data = event.get('Event')
+            if event_data and 'id' in event_data:
+                events_numbers.append({'ID': event_data['id']})
+            else:
+                not_added_counter += 1
+            # If limit exists
+            if limit_int == num:
+                break
+
+        human_readable = tableToMarkdown(f'Total of {len(events_numbers)} events was added to MISP.', events_numbers)
+        if not_added_counter:
+            human_readable = f'{human_readable}\n' \
+                             f'{not_added_counter} events were not added. Might already been added earlier.'
+        return CommandResults(
+            readable_output=human_readable,
+            outputs_prefix='MISP.Event',
+            outputs_key_field='ID',
+            outputs=events_numbers,
+        )
+
+    except ValueError as e:
+        return_error(f'URL [{url}] is not a valid MISP feed. error: {e}')
+
+
+def add_object(event_id: str, obj: MISPObject):
+    """Sending object to MISP and returning outputs
+
+    Args:
+        obj: object to add to MISP
+        event_id: ID of event
+    """
+    response = PYMISP.add_object(event_id, misp_object=obj)
+    if 'errors' in response:
+        return_error(f'Error in `{demisto.command()}` command: {response}')
+    for ref in obj.ObjectReference:
+        response = PYMISP.add_object_reference(ref)
+    formatted_response = replace_keys(response)
+    formatted_response.update({"ID": event_id})
+
+    human_readable = f'Object has been added to MISP event ID {event_id}'
+    return CommandResults(
+        readable_output=human_readable,
+        outputs_prefix='MISP.Event',
+        outputs_key_field='ID',
+        outputs=formatted_response,
+    )
+
+
+def add_file_object(demisto_args: dict = {}):
+    entry_id = demisto_args.get('entry_id')
+    event_id = demisto_args.get('event_id')
+    file_path = demisto.getFilePath(entry_id).get('path')
+    obj = FileObject(file_path)
+    return add_object(event_id, obj)
+
+
+def add_domain_object(demisto_args: dict = {}):
+    """Adds a domain object to MISP
+    domain-ip description: https://www.misp-project.org/objects.html#_domain_ip
+    """
+    text = demisto_args.get('text')
+    event_id = demisto_args.get('event_id')
+    domain = demisto_args.get('name')
+    obj = MISPObject('domain-ip')
+    ips = argToList(demisto_args.get('dns'))
```

Comment:
ips == dns?

Reply:
Changed it to 'ip' in the yml file (it was 'dns' in v2, I didn't notice).